src/components/tl/mlx5/tl_mlx5.h

```diff
@@ -108,19 +108,15 @@ typedef enum
     TL_MLX5_TEAM_STATE_ALLTOALL_INIT,
     TL_MLX5_TEAM_STATE_ALLTOALL_POSTED,
     TL_MLX5_TEAM_STATE_ALLTOALL_READY,
-    TL_MLX5_TEAM_STATE_ALLTOALL_NOT_AVAILABLE
-} ucc_tl_mlx5_team_state_t;
-
-typedef enum
-{
+    TL_MLX5_TEAM_STATE_ALLTOALL_NOT_AVAILABLE,
     TL_MLX5_TEAM_STATE_MCAST_INIT,
     TL_MLX5_TEAM_STATE_MCAST_GRP_JOIN_POST,
     TL_MLX5_TEAM_STATE_MCAST_GRP_JOIN_READY,
     TL_MLX5_TEAM_STATE_MCAST_GRP_JOIN_FAILED,
     TL_MLX5_TEAM_STATE_MCAST_GRP_BCAST_POST,
     TL_MLX5_TEAM_STATE_MCAST_READY,
     TL_MLX5_TEAM_STATE_MCAST_NOT_AVAILABLE
-} ucc_tl_mlx5_team_mcast_state_t;
+} ucc_tl_mlx5_team_state_t;
```

Comment:
if we already use two different field `mcast_state` and `a2a_state` of this enum type, then the enum shouldn't contain `alltoall` or `mcast`

Reply:
_isn't there a problem here if we want, for example, tag both mcast and a2a as "not available"?_
I do not think there is problem. We already clean the team resources if they become unavailable.

_if we already use two different field mcast_state and a2a_state of this enum type, then the enum shouldn't contain alltoall or mcast_
I did not want to have two separate sets of enums for mcast and a2a. This way is easier to maintain. Furthermore, I think It is OK that some of the states in this mlx5 enum set are not applicable to a feature (a2a, mcast, or any future features).

Reply:
> This way is easier to maintain.

Why would it be easier to maintain? With what you propose, there are two field of the same type `a2a_state` and `mcast_state`, but we implicitely assume that each field of the enum is applicable to one and not the other.
In my opinion it is misleading and it adds complexity for no benefit.

> I did not want to have two separate sets of enums for mcast and a2a.

Why not? I would at least remove the duplicates in the enum and some of the explicit mention to MCAST and A2A in the enum. In the current version there is unnecessary complexity. For example, why having `TL_MLX5_TEAM_STATE_MCAST_INIT` `TL_MLX5_TEAM_STATE_MCAST_INIT` and `TL_MLX5_TEAM_STATE_INIT` while they are never used at the same time?

Reply:
In this case, we should not have had ucc_tl_mlx5_team_state_t from the first place. Why did we have ucc_tl_mlx5_team_state_t, while only alltoall related states are stored in it? It was not like tl mlx5 was supposed to be all about alltoall. In this patch, I wanted to maintain the idea that was there from the beginning: having the same mlx5 state machine for all the features including a2a and mcast. 

Based on our discussion, let me remove ucc_tl_mlx5_team_state_t completely and then we will have two separate ucc_tl_mlx5_team_a2a_state_t and ucc_tl_mlx5_team_mcast_state_t instead.

Reply:
Originally, we had one variable `ucc_tl_mlx5_team_state_t` storing the global state of team creation, tracking both both the common parts and the parts specific to alltoall. This state would be increased linearily and that would indicate the stage where we are at. In this case, it make sense to include in the enum all the possible states, also ones containing alltoall or mcast in it. But now there are two independent variables to track the team creation state....

Reply:
Yeah, you're right. Now we have two separate state machines, one for a2a and one for mcast. I think since these two components are independent, it makes sense to keep their state machines separate. what do you think?